src/optimum/intel/openvino/qa_quantize.py

```diff
@@ -0,0 +1,222 @@
+
+    def __getitem__(self, item):
+        """
+        Returns annotation, data and metadata at the specified index.
+        Possible formats:
+        (index, annotation), data
+        (index, annotation), data, metadata
+        """
+        if item >= len(self):
+            raise IndexError
+
+        input_ids = self.features[self.items[item]]["input_ids"]
+        attention_mask = self.features[self.items[item]]["attention_mask"]
+
+        inputs = {"input_ids": input_ids, "attention_mask": attention_mask}       
+        label = self.labels[self.items[item]]
+        return (item, label), inputs
+
+    def prepare_dataset(self):
+        """Prepare dataset"""
+        # Dataset loading
+        self.dataset = load_dataset(self.task, split="validation")
+
+        calibration_features = self.qa_dataloader.get_calib_features(self.dataset)
+
+        for i, feature in enumerate(calibration_features):
+            self.features.append({'input_ids': feature["input_ids"], 'attention_mask': feature["attention_mask"]})
+            self.labels.append({'input_ids': feature["input_ids"], 'answer': feature["answers"], 'id': feature["id"]})
+
+
+class OVQAAccuracyMetric(Metric):
+    # Required methods
+    def __init__(self, config):
+        super().__init__()
+        self.metric_scores = []
+        self.task = config['dataset']['task'].lower()
+        self.max_length = config['dataset']['max_length']
+        self.tokenizer = AutoTokenizer.from_pretrained(config['model_name'])
+        self.metric = load_metric(self.task)
+        self.metric_name = "f1"
+        self.qa_dataloader = QADataLoader(self.tokenizer, self.max_length)
+
+    @property
+    def value(self):
+        """Returns accuracy metric value for the last model output."""
+        return {self.metric_name: self.metric_scores[-1]}
+
+    @property
+    def avg_value(self):
+        """Returns accuracy metric value for all model outputs."""
+        return {self.metric_name: np.ravel(self.metric_scores).mean()}
+
+    def update(self, output, target):
+        """
+        Updates prediction matches.
+
+        :param output: model output
+        :param target: annotations
+        """
+        predictions, references = self.qa_dataloader.postprocess_function(output, target)
+        final_score = self.metric.compute(predictions=predictions, references=references)
+        self.metric_scores.append(final_score[self.metric_name])
+
+    def reset(self):
+        """
+        Resets collected matches
+        """
+        self.metric_scores = []
+
+    def get_attributes(self):
+        """
+        Returns a dictionary of metric attributes {metric_name: {attribute_name: value}}.
+        Required attributes: 'direction': 'higher-better' or 'higher-worse'
+                             'type': metric type
+        """
+        return {self.metric_name: {"direction": "higher-better", "type": "accuracy"}}
+
+class OVQuantizerForQuestionAnswering():

```

Comment:
Most of the code, except DataLoader, can be reused across various tasks, like Text Classification, NER, etc. Please consider making it generic and sharable.